### DISCUSSION TITLE
Email switch not exposed to HA

### DISCUSSION THREAD

**Discussion by cgraun** [fwestenberg/reolink_dev#617] [Q&A]
Hi,
I've read on the documentation that some switches are or not exposed to HA depending on the camera model.

I have an RLC-511WA and only FTP control switch is exposed (not e-mail), which does not allow to disable e-mail alerts when at home.
I can see also IR lights, Push notifications, Record audio, Recording, Siren and Spotlight config switches exposed, although I don't thing they can help.

The idea is to configure everything on the camera (sensibility, lights, SMTP, FTP, etc) and activate or deactivate e-mail alerts and FTP uploads from HA when away or home.

Any idea or suggestion?

Thank you!

**Comment by cpainchaud** [comment]
I think the email switch is broken with latest firmware releases. Someone needs to produce a fix.

**Comment by mnpg** [comment]
As asual,  we (developpers) had to find how the API works with the differents cameras. It's not easy and very boring not to have something stable or to have the same commands workeable on each reolink camera. Each firmware update had lot of of modification and we (developpers) had to follow what's new and without information  (very lack of documentation from reolink).
Not easy at all

**Reply by inch1957** [reply]
Hello :-)
With the http API I can get/set this (and other) info, based on the version from the GetAbility function. So the older versions will use a Get/Set-Emali command and the newer Versions a Get/Set-EmailV20 command. The format of the required parameters is also slightly different but quite well documented in the "Camera HTTP API User Guide_V7.pdf"
Hopefully we'll soon see an implementation here so we can have full integration of most devices :-)
Thanks!!!

**Reply by mnpg** [reply]
> Hello :-) With the http API I can get/set this (and other) info, based on the version from the GetAbility function. So the older versions will use a Get/Set-Emali command and the newer Versions a Get/Set-EmailV20 command. 

Heloo @inch1957 
It depends on the Ability > scheduleVersion (if it is ver=0 : Get/SetEmail works on the camera, if it's ver=1 : Get/SetEmailV20 are the commands to be used )

> The format of the required parameters is also slightly different but quite well documented in the "Camera HTTP API User Guide_V7.pdf" 

Sorry but you're wrong, it depends also about the firmware version. Take a look at my reddit post about the GetEnc command (at the end of the post) -These informations are undocumented in the lastet api guide_v7-

https://www.reddit.com/r/reolinkcam/comments/wnlrnk/looking_for_a_method_to_disable_recording_or/ikrkakf/?context=3

Regards

**Reply by inch1957** [reply]
Thanks for getting back to me and for your precise details re firmware. What (I believe) is most expected is the email on/off function like the FTP on/off that is already available on (most) cameras). I personally am ok with waiting for this to be integrated, in the meantime I've written automations and scripts (this however is not as efficient as an integration...).
Great job, thanks!!! :-)
